sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -469,6 +475,30 @@ public static string CanonicalizeResourcePathV2(Uri endpoint, string resourcePat
             return canonicalizedResourcePath;
         }
 
+        /// <summary>
+        /// This method returns the full canonicalized resource path which includes the resource path from the request and the resource path from the endpoint.
+        /// This is just a helper method to be used internally.
+        /// </summary>
+        /// <param name="encodedSegments">If double encoded, encoded segments contains the encoded segments from the resource path from the request</param>
+        /// <param name="resourcePathFromEndpoint">This is the resource path that comes from the endpoint itself and not the request</param>
+        /// <returns>The full canonicalized resource path that contains the single encoded canonicalized resource path from the endpoint and the double encoded
+        /// resource path from the request</returns>
+        private static string GetFullCanonicalizedResourcePath(IEnumerable<string> encodedSegments, string resourcePathFromEndpoint)
```

Comment:
nit: I would put `resourcePathFromEndpoint` as the first parameter since it is the first part of the path.